diff --git a/Makefile b/Makefile
index 21262b8..0d6a986 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,6 @@
 export ARCHS = arm64 arm64e
-export TARGET = iphone:clang:13.3.1
+export TARGET = iphone:clang:14.5
+export THEOS_PACKAGE_SCHEME = rootless
 
 INSTALL_TARGET_PROCESSES = SpringBoard
 SUBPROJECTS += Tweak Preferences

diff --git a/control b/control
index 1cfb26a..2bb4b5b 100644
--- a/control
+++ b/control
@@ -1,6 +1,6 @@
 Package: love.litten.crystal
 Name: Crystal 💎
-Depends: mobilesubstrate, preferenceloader, ws.hbang.common (>= 1.14), firmware (>= 13.0)
+Depends: mobilesubstrate, preferenceloader, ws.hbang.common (>= 1.14), firmware (>= 15.0)
 Version: 1.4.1
 Architecture: iphoneos-arm
 Description: Headphone listening mode automations

diff --git a/Preferences/CSTRootListController.m b/Preferences/CSTRootListController.m
index c01560f..cc7dfe8 100644
--- a/Preferences/CSTRootListController.m
+++ b/Preferences/CSTRootListController.m
@@ -42,7 +42,7 @@ - (void)viewDidLoad {
 
     self.iconView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 10, 10)];
     [[self iconView] setContentMode:UIViewContentModeScaleAspectFit];
-    [[self iconView] setImage:[UIImage imageWithContentsOfFile:@"/Library/PreferenceBundles/CrystalPreferences.bundle/icon.png"]];
+    [[self iconView] setImage:[UIImage imageWithContentsOfFile:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/icon.png"]];
     [[self iconView] setAlpha:0];
     [[[self navigationItem] titleView] addSubview:[self iconView]];
 
@@ -62,7 +62,7 @@ - (void)viewDidLoad {
     self.headerView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 200, 200)];
     self.headerImageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 200, 200)];
     [[self headerImageView] setContentMode:UIViewContentModeScaleAspectFill];
-    [[self headerImageView] setImage:[UIImage imageWithContentsOfFile:@"/Library/PreferenceBundles/CrystalPreferences.bundle/Banner.png"]];
+    [[self headerImageView] setImage:[UIImage imageWithContentsOfFile:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/Banner.png"]];
     [[self headerImageView] setClipsToBounds:YES];
     [[self headerView] addSubview:[self headerImageView]];
 
@@ -74,7 +74,7 @@ - (void)viewDidLoad {
         [self.headerImageView.bottomAnchor constraintEqualToAnchor:self.headerView.bottomAnchor],
     ]];
 
-    if ([[NSFileManager defaultManager] fileExistsAtPath:@"/Library/MobileSubstrate/DynamicLibraries/Crystal.disabled"]) {
+    if ([[NSFileManager defaultManager] fileExistsAtPath:@"/var/jb/Library/MobileSubstrate/DynamicLibraries/Crystal.disabled"]) {
         [[self enableSwitch] setEnabled:NO];
 
         UIAlertController* alertController = [UIAlertController alertControllerWithTitle:@"Crystal" message:@"Crystal has detected that you have disabled it with iCleaner Pro, here are some quick actions you can perform" preferredStyle:UIAlertControllerStyleAlert];
@@ -96,7 +96,7 @@ - (void)viewDidLoad {
         [alertController addAction:ignoreAction];
 
         [self presentViewController:alertController animated:YES completion:nil];
-    } else if ([[NSFileManager defaultManager] fileExistsAtPath:@"/Library/MobileSubstrate/DynamicLibraries/DayNightSwitch.dylib"]) {
+    } else if ([[NSFileManager defaultManager] fileExistsAtPath:@"/var/jb/Library/MobileSubstrate/DynamicLibraries/DayNightSwitch.dylib"]) {
         UIAlertController* alertController = [UIAlertController alertControllerWithTitle:@"Crystal" message:@"Crystal has detected that you have DayNightSwitch installed, which causes issues with Crystal's preferences\n\n To continue, please disable DayNightSwitch with iCleaner Pro or uninstall it temporarily" preferredStyle:UIAlertControllerStyleAlert];
 
         UIAlertAction* confirmAction = [UIAlertAction actionWithTitle:@"Okey" style:UIAlertActionStyleDestructive handler:^(UIAlertAction * action) {
@@ -221,7 +221,7 @@ - (void)resetPreferences {
     for (NSString* key in [[self preferences] dictionaryRepresentation]) {
         if (![key isEqualToString:@"wasWelcomed"]) [[self preferences] removeObjectForKey:key];
     }
-    [[NSFileManager defaultManager] removeItemAtPath:@"/var/mobile/Library/Preferences/love.litten.crystalpreferences/" error:nil];
+    [[NSFileManager defaultManager] removeItemAtPath:@"/var/jb/var/mobile/Library/Preferences/love.litten.crystalpreferences.plist" error:nil];
     
     [[self enableSwitch] setOn:NO animated:YES];
     [self respring];
@@ -237,7 +237,7 @@ - (void)respring {
     [UIView animateWithDuration:1 delay:0 options:UIViewAnimationOptionCurveEaseOut animations:^{
         [[self blurView] setAlpha:1];
     } completion:^(BOOL finished) {
-        if (![[NSFileManager defaultManager] fileExistsAtPath:@"/Library/MobileSubstrate/DynamicLibraries/shuffle.dylib"])
+        if (![[NSFileManager defaultManager] fileExistsAtPath:@"/var/jb/Library/MobileSubstrate/DynamicLibraries/shuffle.dylib"])
             [HBRespringController respringAndReturnTo:[NSURL URLWithString:@"prefs:root=Crystal"]];
         else
             [HBRespringController respringAndReturnTo:[NSURL URLWithString:@"prefs:root=Tweaks&path=Crystal"]];

diff --git a/Preferences/WelcomeViewController.m b/Preferences/WelcomeViewController.m
index 1edce0d..c8629e5 100644
--- a/Preferences/WelcomeViewController.m
+++ b/Preferences/WelcomeViewController.m
@@ -8,7 +8,7 @@ - (void)viewDidLoad {
     [[self view] setBackgroundColor:[UIColor systemBackgroundColor]];
 
 
-    NSData* inData = [NSData dataWithContentsOfURL:[NSURL fileURLWithPath:@"/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/Circle Of Love.ttf"]];
+    NSData* inData = [NSData dataWithContentsOfURL:[NSURL fileURLWithPath:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/Circle Of Love.ttf"]];
     CFErrorRef error;
     CGDataProviderRef provider = CGDataProviderCreateWithCFData((CFDataRef)inData);
     CGFontRef font = CGFontCreateWithDataProvider(provider);
@@ -23,7 +23,7 @@ - (void)viewDidLoad {
     // header
     self.headerImageView = [UIImageView new];
     [[self headerImageView] setContentMode:UIViewContentModeScaleAspectFill];
-    [[self headerImageView] setImage:[UIImage imageWithContentsOfFile:@"/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/header.png"]];
+    [[self headerImageView] setImage:[UIImage imageWithContentsOfFile:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/header.png"]];
     [[self view] addSubview:[self headerImageView]];
 
     [[self headerImageView] setTranslatesAutoresizingMaskIntoConstraints:NO];
@@ -90,7 +90,7 @@ - (void)viewDidLoad {
     self.twitterCellIcon = [UIButton new];
     [[self twitterCellIcon] addTarget:self action:@selector(openTwitterURL) forControlEvents:UIControlEventTouchUpInside];
     [[self twitterCellIcon] setContentMode:UIViewContentModeScaleAspectFill];
-    [[self twitterCellIcon] setImage:[UIImage imageWithContentsOfFile:@"/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/twitter.png"] forState:UIControlStateNormal];
+    [[self twitterCellIcon] setImage:[UIImage imageWithContentsOfFile:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/twitter.png"] forState:UIControlStateNormal];
     [[self twitterCellIcon] setClipsToBounds:YES];
     [[[self twitterCellIcon] layer] setCornerRadius:10];
     [[[self twitterCellIcon] layer] setBorderColor:[[UIColor colorWithRed:0.93 green:0.76 blue:1 alpha:1] CGColor]];
@@ -142,7 +142,7 @@ - (void)viewDidLoad {
     self.githubCellIcon = [UIButton new];
     [[self githubCellIcon] addTarget:self action:@selector(openGitHubURL) forControlEvents:UIControlEventTouchUpInside];
     [[self githubCellIcon] setContentMode:UIViewContentModeScaleAspectFill];
-    [[self githubCellIcon] setImage:[UIImage imageWithContentsOfFile:@"/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/github.png"] forState:UIControlStateNormal];
+    [[self githubCellIcon] setImage:[UIImage imageWithContentsOfFile:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/github.png"] forState:UIControlStateNormal];
     [[self githubCellIcon] setClipsToBounds:YES];
     [[[self githubCellIcon] layer] setCornerRadius:10];
     [[[self githubCellIcon] layer] setBorderColor:[[UIColor colorWithRed:0.84 green:0.89 blue:1 alpha:1] CGColor]];
@@ -194,7 +194,7 @@ - (void)viewDidLoad {
     self.discordCellIcon = [UIButton new];
     [[self discordCellIcon] addTarget:self action:@selector(openDiscordURL) forControlEvents:UIControlEventTouchUpInside];
     [[self discordCellIcon] setContentMode:UIViewContentModeScaleAspectFill];
-    [[self discordCellIcon] setImage:[UIImage imageWithContentsOfFile:@"/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/discord.png"] forState:UIControlStateNormal];
+    [[self discordCellIcon] setImage:[UIImage imageWithContentsOfFile:@"/var/jb/Library/PreferenceBundles/CrystalPreferences.bundle/welcome/discord.png"] forState:UIControlStateNormal];
     [[self discordCellIcon] setClipsToBounds:YES];
     [[[self discordCellIcon] layer] setCornerRadius:10];
     [[[self discordCellIcon] layer] setBorderColor:[[UIColor colorWithRed:0.58 green:0.66 blue:0.71 alpha:1] CGColor]];

